iree/samples/variables_and_state/variables_and_state.ipynb

```diff
@@ -0,0 +1,471 @@
+        "vm_module = ireert.VmModule.from_flatbuffer(flatbuffer_blob)\n",
+        "backend = module_utils.BackendInfo(\"iree_vmvx\")\n",
+        "config = ireert.Config(backend.driver)\n",
+        "ctx = ireert.SystemContext(config=config)\n",
+        "ctx.add_vm_module(vm_module)\n",
+        "\n",
+        "# Our @tf.functions are accessible by name on the module named 'module'\n",
+        "print(ctx.modules.module)\n",
+        "get_value_f = ctx.modules.module[\"get_value\"]\n",
+        "set_value_f = ctx.modules.module[\"set_value\"]\n",
+        "add_to_value_f = ctx.modules.module[\"add_to_value\"]\n",
+        "reset_value_f = ctx.modules.module[\"reset_value\"]"
+      ],
+      "execution_count": 6,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "<BoundModule <VmModule module : [add_to_value$async, add_to_value$sync, get_value$async, get_value$sync, reset_value$async, reset_value$sync, set_value$async, set_value$sync, __init]>>\n"
+          ],
+          "name": "stdout"
+        },
+        {
+          "output_type": "stream",
+          "text": [
+            "Created IREE driver vmvx: <iree.runtime.binding.HalDriver object at 0x7ffa89e6ecb0>\n",
+            "SystemContext driver=<iree.runtime.binding.HalDriver object at 0x7ffa89e6ecb0>\n"
+          ],
+          "name": "stderr"
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "CQffg1iQatkb",
+        "outputId": "4cb5398b-8691-45ab-c378-941c46c12416"
+      },
+      "source": [
+        "print(get_value_f())\n",
+        "set_value_f(101)\n",
+        "print(get_value_f())\n",
+        "\n",
+        "add_to_value_f(20)\n",
+        "print(get_value_f())\n",
+        "add_to_value_f(-50)\n",
+        "print(get_value_f())\n",
+        "\n",
+        "reset_value_f()\n",
+        "print(get_value_f())"
+      ],
+      "execution_count": 7,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "0\n",
+            "101\n",
+            "121\n",
+            "71\n",
+            "0\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "wCvwX1IEokm6"
+      },
+      "source": [
+        "## Download compilation artifacts"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 102
+        },
+        "id": "bUaNUkS2ohRj",
+        "outputId": "73068e43-df6f-49f4-8f39-725fb1971901"
+      },
+      "source": [
+        "ARTIFACTS_ZIP = \"/tmp/variables_and_state_colab_arficats.zip\"\n",
+        "\n",
+        "print(f\"Zipping '{ARTIFACTS_DIR}' to '{ARTIFACTS_ZIP}' for download...\")\n",
+        "!zip -r {ARTIFACTS_ZIP} {ARTIFACTS_DIR}\n",
```

Comment:
this can be `!cd {ARTIFACTS_DIR} && zip -r {ARTIFACTS_ZIP} .` to avoid having the annoying `/tmp/iree/colab_artifact` directory tree inside the zip
![image](https://user-images.githubusercontent.com/75337/120745695-8e90a900-c4b2-11eb-9cff-42263f6b8039.png)